src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/DependencyGraphSpecRequestProvider.cs

```diff
@@ -63,7 +63,7 @@ private IReadOnlyList<RestoreSummaryRequest> GetRequestsFromItems(RestoreArgs re
             {
                 var closure = dgFile.GetClosure(projectNameToRestore);
 
-                var projectDependencyGraphSpec = dgFile.WithProjectClosure(projectNameToRestore);
```

Comment:
Cloning in VS might actually lead to inconsistency, did you look into that? 

If VS has no need for the clones, then we just remove the cloning completely, but last time I investigated that wasn't the case.

Reply:
Also I don't want package source updates of transitive projects to affect the no-op hash for the parent projects.

Reply:
I'm going to add a property to DependencyGraphSpec instead which will guarantee that I don't break existing code paths